jit.js

```diff
@@ -287,20 +478,22 @@ to single-step.
                     break;
                 // send literal selector
                 case 0xD0: case 0xD8:
-                    this.generateSend("lit[", 1 + (byte & 0x0F), "]", 0, false);
+                    this.generateSend("lit[", 1 + (byte & 0x0F), "]", 0, false, this.method.ic[this.pc]);
                     break;
                 case 0xE0: case 0xE8:
-                    this.generateSend("lit[", 1 + (byte & 0x0F), "]", 1, false);
+                    this.generateSend("lit[", 1 + (byte & 0x0F), "]", 1, false, this.method.ic[this.pc]);
                     break;
                 case 0xF0: case 0xF8:
-                    this.generateSend("lit[", 1 + (byte & 0x0F), "]", 2, false);
+                    this.generateSend("lit[", 1 + (byte & 0x0F), "]", 2, false, this.method.ic[this.pc]);
                     break;
             }
         }
         var funcName = this.functionNameFor(optClass, optSel);
         if (this.singleStep) {
             if (this.debug) this.source.push("// all valid PCs have a label;\n");
+            this.labelBootstrapEnd = this.source.length;
             this.source.push("default: throw Error('invalid PC'); }"); // all PCs handled
+            this.deleteUnneededLabelBootstrap();
             return new Function("return function " + funcName + "(vm, singleStep) {\n" + this.source.join("") + "\n}")();
```

Comment:
This does not belong in the singleStep branch. That one is used only while debugging (in the Lively debugger). During normal execution, `singleStep` is false. When single-stepping we actually need all the labels (unless the method only has a single bytecode, which I don't think exists). The way you wrote it, this never actually gets used (look at the generated code for a simple method, e.g. `SqueakJS.vm.findMethod("Rectangle>>setOrigin:corner:").compiled` ).